File: js/palindrome.js
function palindrome(word) {
    let cleanWord = word.toString().toLowerCase().replace(/[^a-zA-Z0-9]/g, '')
    let first_half = '';
    let second_half = '';

    if (word.length % 2 !== 0) {
        first_half = cleanWord.slice(0, Math.floor(cleanWord.length / 2))
        second_half = cleanWord.slice(Math.ceil(cleanWord.length / 2)).split('').reverse().join('')
    } else {
        first_half = cleanWord.slice(0, cleanWord.length / 2)
        second_half = cleanWord.slice((cleanWord.length + 1) / 2).split('').reverse().join('')        
    }   
    return first_half === second_half ? true : false
};

module.exports = {palindrome};

File: python/palindrome.py
import re, math

def palindrome(word):
    clean_word = re.sub(r"[^a-zA-Z0-9]", "", str(word).lower())
    
    first_half = ''
    second_half = ''

    if len(clean_word) % 2 != 0:
        half = math.floor(len(clean_word)/2)
        first_half = clean_word[0 : half]
        second_half = clean_word[half + 1 : ][::-1]
    else :
        half = int(len(clean_word) / 2)
        first_half = clean_word[0 : half]
        second_half = clean_word[half:][::-1]
    return True if first_half == second_half  else False

